codegen/gql_code_builder/lib/src/when_extension.dart

```diff
@@ -40,9 +40,15 @@ Extension? inlineFragmentWhenExtension(
     {required String baseTypeName,
     required List<InlineFragmentNode> inlineFragments,
     required InlineFragmentSpreadWhenExtensionConfig config,
+    required Map<String, Set<String>> possibleTypesMap,
     required Map<String, Reference> dataClassAliasMap}) {
   final inlineFragmentsWithTypConditions = inlineFragments
-      .where((inlineFragment) => inlineFragment.typeCondition != null)
+      .where((inlineFragment) =>
+          inlineFragment.typeCondition != null
+          // except interfaces on when extension
+          &&
+          !possibleTypesMap
+              .containsKey(inlineFragment.typeCondition!.on.name.value))
```

Comment:
@knaeckeKami And I just checked WhenExtension codes too, And it seems okay. There would be no changes without feature flag, because added condition [line 49-51] will be always true when the flag is off.